user/serializers.py

```diff
@@ -2,0 +1,28 @@
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "email", "password", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {
+            "password": {"write_only": True, "min_length": 5},
+        }
+
+    def create(self, validated_data):
+        """ Create user with encrypted password and return it. """
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        """ Update user with encrypted password and return it. """
+        password = validated_data.pop("password", None)
+        user = super().update(instance, validated_data)
```

Comment:
In the `update` method, the password is popped from `validated_data` and set separately. This is a good practice to ensure the password is hashed. Ensure that any other sensitive fields are handled similarly.